sklearn/utils/_test_common/instance_generator.py

```diff
@@ -566,38 +587,58 @@
     PLSRegression: {"check_dict_unchanged": dict(max_iter=5, n_components=1)},
     PLSSVD: {"check_dict_unchanged": dict(n_components=1)},
     PoissonRegressor: {
-        "check_sample_weight_equivalence": [
+        "check_sample_weight_equivalence_on_dense_data": [
             dict(solver="newton-cholesky"),
             dict(solver="lbfgs"),
-        ]
+        ],
+        "check_sample_weight_equivalence_on_sparse_data": [
+            dict(solver="newton-cholesky"),
+            dict(solver="lbfgs"),
+        ],
     },
     PolynomialCountSketch: {"check_dict_unchanged": dict(n_components=1)},
     QuantileRegressor: {
-        "check_sample_weight_equivalence": [
+        "check_sample_weight_equivalence_on_dense_data": [
             dict(quantile=0.5),
             dict(quantile=0.75),
             dict(solver="highs-ds"),
             dict(solver="highs-ipm"),
-        ]
+        ],
+        "check_sample_weight_equivalence_on_sparse_data": [
+            dict(quantile=0.5),
+            dict(quantile=0.75),
+            dict(solver="highs-ds"),
+            dict(solver="highs-ipm"),
+        ],
     },
     RBFSampler: {"check_dict_unchanged": dict(n_components=1)},
     Ridge: {
-        "check_sample_weight_equivalence": [
+        "check_sample_weight_equivalence_on_dense_data": [
             dict(solver="svd"),
             dict(solver="cholesky"),
             dict(solver="sparse_cg"),
             dict(solver="lsqr"),
             dict(solver="lbfgs", positive=True),
-        ]
+        ],
+        "check_sample_weight_equivalence_on_sparse_data": [
+            dict(solver="sparse_cg"),
+            dict(solver="lsqr"),
+            dict(solver="lbfgs", positive=True),
```

Comment:
I think solvers like `"lbfgs"` have no specific code path for dense vs sparse. In those cases, I think we should refrain to duplicate the configs between `check_sample_weight_equivalence_on_dense_data` and `check_sample_weight_equivalence_on_sparse_data`.

Reply:
I think the same (no specific code path for dense vs sparse) holds for:
- GLM like `TweedieRegressor`, `PoissonRegressor` and `GammaRegressor` for both solvers "newton-cholesky" and "lbfgs".
- `QuantileRegressor` 
- `LogisticRegression` for solvers "lbfgs", "newton-cg" and "newton-cholesky". 

However I think we have different code paths for:
- `DecisionTreeRegressor/Classifier`
- `StandardScaler`
- `LogisticRegression` for solver "liblinear"
- `Ridge/RidgeClassifier` with solver "sparse_cg" or "lsqr". Maybe solver="lbfgs" with fit_intercept=True could be considered a different code path as X_offset is passed  only when X is sparse.